p2p/rpc/rpcserver.go

```diff
@@ -176,3 +185,86 @@ func (p RPCServer) GetSettings(ctx context.Context, in *empty.Empty) (*pb.P2PSet
 	defer p.lock.Unlock()
 	return &pb.P2PSettings{}, nil
 }
+
+// SendDirectMessage sends a direct message
+func (p RPCServer) SendDirectMessage(ctx context.Context, in *pb.SendDirectMessageRequest) (*empty.Empty, error) {
+	id, err := p2p.StringToID(in.PeerID)
+	if err != nil {
+		return nil, err
+	}
+	peer := p.hostNode.FindPeerByID(id)
+	if peer == nil {
+		return nil, errors.New("Can't find peer by ID")
+	}
+
+	message, err := p2p.BytesToMessage(in.GetMessage())
+	if err != nil {
+		return nil, err
+	}
+
+	peer.SendMessage(message)
+
+	return nil, nil
+}
+
+// SubscribeDirectMessage subscribes a direct message
+func (p RPCServer) SubscribeDirectMessage(ctx context.Context, in *pb.SubscribeDirectMessageRequest) (*pb.DirectMessageSubscription, error) {
+	p.lock.Lock()
+	defer p.lock.Unlock()
+	subID := *p.currentSubID
+	*p.currentSubID++
+
+	subChan := make(chan pb.ListenForDirectMessagesResponse)
+	p.directMessageSubChannels[subID] = subChan
+	p.directMessageCancelChannels[subID] = make(chan bool)
+
+	s := p.hostNode.RegisterMessageHandler(in.MessageName, func(peer *p2p.PeerNode, message proto.Message) {
```

Comment:
include peerID here

Reply:
Currently host node doesn't distinguish between peers when dispatching direct message because it's quite cheap to just listen to message from every peers in p2p itself. The message-per-PeerID is handled in RPC server.